icechunk-python/src/lib.rs

```diff
@@ -88,10 +88,61 @@ fn initialize_logs(py: Python) -> PyResult<()> {
     Ok(())
 }
 
+fn extract_module_names(filter: &str) -> Vec<&str> {
+    filter
+        .split(',')
+        .map(|part| part.trim())
+        .filter_map(|part| {
+            if part.contains(':') {
+                Some(part.split(':').next().unwrap_or(""))
+            } else {
+                None
+            }
+        })
+        .filter(|module| !module.is_empty())
+        .collect()
+}
+
+fn is_likely_icechunk_misspelling(module: &str) -> bool {
+    const GENERIC_LOG_LEVELS: &[&str] = &["trace", "debug", "info", "warn", "error"];
+
+    if GENERIC_LOG_LEVELS.contains(&module) {
+        return false;
+    }
+
+    let base_module = module.split("::").next().unwrap_or(module);
+    let distance = strsim::levenshtein("icechunk", base_module);
+
+    distance > 0 && distance <= 2 && base_module.len() >= 6
+}
+
+fn check_filter_for_misspellings(filter: &str) {
+    let modules = extract_module_names(filter);
+
+    for module in modules {
+        if is_likely_icechunk_misspelling(module) {
+            let misspelled = module.split("::").next().unwrap_or(module);
+            // Sanitize the user input to prevent ANSI injection attacks
+            let sanitized =
+                misspelled.chars().filter(|c| !c.is_control()).collect::<String>();
+
+            eprintln!(
+                "\x1b[93m\x1b[1mWarning\x1b[0m: Did you mean '\x1b[92m\x1b[1micechunk\x1b[0m' instead of '\x1b[91m{}\x1b[0m' in log filter?",
```

Comment:
Let's not do colors. Otherwise, we need to respect the NO_COLOR variable, verify terminal type, etc

Reply:
done